contrib/platform/src/com/sun/jna/platform/mac/CoreFoundation.java

```diff
@@ -0,0 +1,1029 @@
+         * Index for the type of {@link CFNumberRef} stored.
+         *
+         * @return a {@link CFIndex} representing the enum ordinal.
+         */
+        public CFIndex typeIndex() {
+            return new CFIndex(this.ordinal());
+        }
+    }
+
+    /**
+     * A reference to a {@code CFBoolean} object.
+     */
+    class CFBooleanRef extends CFTypeRef {
+        public CFBooleanRef() {
+            super();
+        }
+
+        public CFBooleanRef(Pointer p) {
+            super(p);
+            if (!isTypeID(BOOLEAN_TYPE_ID)) {
+                throw new ClassCastException("Unable to cast to CFBoolean. Type ID: " + getTypeID());
+            }
+        }
+
+        /**
+         * Convert a reference to a Core Foundations Boolean into its {@code boolean}
+         *
+         * @return The corresponding {@code boolean}
+         */
+        public boolean booleanValue() {
+            return 0 != INSTANCE.CFBooleanGetValue(this);
+        }
+    }
+
+    /**
+     * A reference to an immutable {@code CFArray} object.
+     * <p>
+     * CFArray is “toll-free bridged” with its Cocoa Foundation counterpart,
+     * {@code NSArray}. Therefore, in a method where you see an {@code NSArray *}
+     * parameter, you can pass in a {@link #CFArrayRef}.
+     */
+    class CFArrayRef extends CFTypeRef {
+        public CFArrayRef() {
+            super();
+        }
+
+        public CFArrayRef(Pointer p) {
+            super(p);
+            if (!isTypeID(ARRAY_TYPE_ID)) {
+                throw new ClassCastException("Unable to cast to CFArray. Type ID: " + getTypeID());
+            }
+        }
+
+        /**
+         * Convenience method for {@link #CFArrayGetCount} on this object
+         *
+         * @return The number of values in this array.
+         */
+        public CFIndex getCount() {
+            return INSTANCE.CFArrayGetCount(this);
+        }
+
+        /**
+         * Convenience method for {@link #CFArrayGetValueAtIndex} on this object
+         *
+         * @param idx
+         *            The index of the value to retrieve.
+         * @return The value at the {@code idx} index.
+         */
+        public Pointer getValueAtIndex(CFIndex idx) {
+            return INSTANCE.CFArrayGetValueAtIndex(this, idx);
+        }
+    }
+
+    /**
+     * A reference to an immutable {@code CFData} object.
+     */
+    class CFDataRef extends CFTypeRef {
+        public CFDataRef() {
+            super();
+        }
+
+        public CFDataRef(Pointer p) {
+            super(p);
+            if (!isTypeID(DATA_TYPE_ID)) {
+                throw new ClassCastException("Unable to cast to CFData. Type ID: " + getTypeID());
+            }
+        }
+
+        /**
+         * Convenience method for {@link #CFDataGetLength} on this object
+         *
+         * @return An index that specifies the number of bytes associated with this
+         *         object.
+         */
+        public CFIndex getLength() {
```

Comment:
Same comment as in the array case.

Reply:
And same answer, with `int` preferred as this deals with the length/index of a `byte[]`.